src/frontend/sharedComponents/LeaveProjectModalContent/LeaveProject.tsx

```diff
@@ -70,46 +71,38 @@ export const LeaveProject = ({
   const [error, setError] = React.useState(false);
   const [isChecked, setIsChecked] = React.useState(false);
   const leaveProject = useLeaveProject();
-  const [combinedLoading, setCombinedLoading] = React.useState(false);
+  const {projectId} = useActiveProject();
 
   function handleLeavePress() {
     if (!isChecked) {
       setError(true);
       return;
     }
-    setCombinedLoading(true);
-    // we want to accept first because the invitor will be able to cancel. this avoids the user leaving a project, and then their invite being cancelled before they were able to join.
+    // we want to accept first because the invitor will be able to cancel.
+    // this avoids the user leaving a project, and then their invite being cancelled before they were able to join.
     accept.mutate(
       {inviteId},
       {
         onSuccess: () => {
-          closeSheet();
-          setCombinedLoading(false);
+          leaveProject.mutate(projectId, {
+            onSuccess: () => {
+              closeSheet();
+            },
+            onError: err => {
+              Sentry.captureException(err);
+            },
+          });
         },
-        // This is commented out for now and issue created: https://github.com/digidem/comapeo-mobile/issues/525
-        // onSuccess: () => {
-        //   leaveProject.mutate(undefined, {
-        //     onSuccess: () => {
-        //       closeSheet();
-        //       setCombinedLoading(false);
-        //     },
-        //     onError: err => {
-        //       Sentry.captureException(err)
-        //       setCombinedLoading(false);
-        //     },
-        //   });
-        // },
         onError: err => {
           Sentry.captureException(err);
-          setCombinedLoading(false);
         },
       },
     );
   }
 
   return (
     <>
-      {combinedLoading ? (
+      {accept.status === 'pending' || leaveProject.status === 'pending' ? (
```

Comment:
have you tested this out? I think this actually will still cause it to flicker becasue there is a breif time when `accept.status === success` but the `leaveProject` mutaion hasnt been started yet, so `leaveProject.status === idle`. When you made [this comment ](https://github.com/digidem/comapeo-mobile/pull/526#discussion_r1683485870) i think we both were thinking of `query`, where `pending===true` when the query hasn't been made yet. But in a mutation `pending===false` when it hasnt been called yet

Reply:
hm i see what you mean. haven't tested it enough to confirm but what you describe makes sense. i was considering doing something like the following so that we could just rely on the `accept.status`:

```ts
accept.mutate({
  onSuccess: async () => {
    await leaveProject.mutateAsync({ ... })
  }
})
```

in that case, the accept mutation will stay pending until `leaveProject.mutateAsync()` finishes. was a little worried that it might result in some edge cases around error-handling though

Reply:
although part of me thinks that what i currently have here works as intended. would have to do some logging to confirm, but my basic understanding is that the following sequence happens:

1. Start accept mutation, so `accept.status === 'pending'`

2. The accept mutation succeeds, so `onSuccess` is called, which calls the leave project mutation so `leave.status === 'pending'`

3. After `accept.onSuccess()` finishes, `accept.status === 'idle'`, but since leave project mutation was kicked off in (2), we have `accept.status === 'idle'` and `leave.status === 'pending`.

Reply:
yeah this is what i get in the logs (although replace `idle` with `success` since they're mutations i guess):

```sh
 LOG  {"accept": "pending", "leave": "idle"}
 WARN  [MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 stopped listeners added. Use emitter.setMaxListeners() to increase limit]
 WARN  [MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 internal-error listeners added. Use emitter.setMaxListeners() to increase limit]
 LOG  {"accept": "pending", "leave": "pending"}
{"accept": "success", "leave": "pending"}
 {"accept": "success", "leave": "pending"}
 {"accept": "success", "leave": "success"}
{"accept": "success", "leave": "success"}
{"accept": "success", "leave": "success"}
```

Reply:
I must have done something slightly different because i originally was just using the loading state of the 2 mutations in the conditional rendering statement, and it was causing a flicker, which is why i created that `combinedLoadingState`But it seems like its not a problem. Maybe ill just test it out with a device really quick to make sure

Reply:
With current check:

https://github.com/user-attachments/assets/9c067d86-8d70-445f-830c-98281e844808

---

After adding condition for when they're both in  `success` status:

https://github.com/user-attachments/assets/721902ae-c392-4f90-8e1c-a3ab70727405


I think there's a slight improvement so probably worth adding the additional check. There's additional flickering going on in the map but that's due to lots of query invalidation that's outside the scope of this PR

Reply:
see ff1982ff9d1e0b6d60890eb06e37930acc64dad7 for relevant commit

Reply:
ahhh I see now. So the 'combinedLoading' reduced the flickering not because of a race condition, but because it was set AFTER the modal was already being set to close